### PR TITLE
Add str pure methods + tests

### PR DESCRIPTION
**Pull request by travcunn**

### CODE REVIEW

File: test_fatoptimizer.py [hunk 1]
```diff
@@ -3206,6 +3206,70 @@ def test_str_encode(self):
...
+
+    def test_str_capitalize(self):
+        self.check_optimize(r'"abc".capitalize()', '"Abc"')
+
+    def test_str_swapcase(self):
+        self.check_optimize(r'"aBc".swapcase()', '"AbC"')
+
+    def test_str_casefold(self):
+        self.check_optimize(r'"ABC".casefold()', '"abc"')
+        self.check_optimize(r'"ß".casefold()', '"ss"')
```
Comment (haypo) [inline test_fatoptimizer.py]: Please fold these case functions into a single test function. One test perf method is enough, fatoptimizer test suite is not the Python test suite!

File: test_fatoptimizer.py [hunk 2]
```diff
@@ -3206,6 +3206,70 @@ def test_str_encode(self):
...
+        self.check_optimize(r'"\u00B2".isnumeric()', 'True')
+        self.check_optimize(r'"\u00BD".isnumeric()', 'True')
+        self.check_optimize(r'"1.0".isnumeric()', 'False')
+        self.check_optimize(r'"1".isnumeric()', 'True')
+
+    def test_str_isupper(self):
+        self.check_optimize(r'"".isupper()', 'False')
+        self.check_optimize(r'"ABC".isupper()', 'True')
+        self.check_optimize(r'"aBc".isupper()', 'False')
+        self.check_optimize(r'"abc".isupper()', 'False')
```
Comment (haypo) [inline test_fatoptimizer.py]: Same here: fold is_xxx() into a single test function and restrict to one test per method.

**Comment by haypo** [comment]
Can you please fold your 13 commits into a single commit? I don't think that it's worth to have so many small commits.

Anyway, it's cool to get contributions, thanks :-)

File: test_fatoptimizer.py (hunk 1 above)
Comment (travcunn) [inline test_fatoptimizer.py]: Okay I wasn't sure about how to go about tests here. Thanks for the clarification.

**Comment by travcunn** [comment]
Created a new PR: #9

**Pull request closed without merging**